metaflow/plugins/argo/argo_workflows_cli.py

```diff
@@ -484,13 +485,14 @@ def trigger(obj, run_id_file=None, **kwargs):
     def _convert_value(param):
         # Swap `-` with `_` in parameter name to match click's behavior
         val = kwargs.get(param.name.replace("-", "_").lower())
-        return (
-            json.dumps(val)
-            if param.kwargs.get("type") == JSONType
-            else val()
-            if callable(val)
-            else val
-        )
+        val = kwargs.get(param.name.replace("-", "_").lower())
+        if param.kwargs.get("type") == JSONType:
+            val = json.dumps(val)
+        elif isinstance(param.kwargs.get("type"), FilePathClass):
```

Comment:
If you are asking for the original name of the local file, I don't think I can do that (well, I can add it to the JSON blob if you want but can't replace it wholesale with that since that local file path would make no sense on a remote system.

If you are talking about just the url portion, I *think* I could swing it with two (BIG) caveats in my mind:
 - we would lose the is_text and encoding information (and size but that is less of a problem -- it can be recovered)
 - we would definitely need to support the s3:// type of include files.

IMHO, keeping it as is (and maybe adding something to the JSON to make it clearer to the user what it is (for example: "note": "Representation of user included file 'foobar.txt'" may be better.

In terms of the UI allowing modifications, I don't know enough about Argo (or any of the others) to comment. I can tell you how we would do it on maestro though (they have various parameter modes).

Reply:
I added a "note" field in the blurb.